### PR TITLE
dotCMS/core#22566 fix Improvements for ENTER key event

### LINKED ISSUE #22566: Improvements for ENTER key event
The following improvement suggestions came after this change https://github.com/dotCMS/core/issues/19158

- Create Templates (Next Button does not react to ENTER):

178408093-ec79d543-61b8-4cd4-ae18-ceb262f00f28

- Apps / Import (The main dialog behaves correctly but once a file has been chosen and an invalid password is submitted) You'll see an invalid request dialog. Which does not react to ENTER even after having set the focus on it. The only way to dismiss the dialog is by clicking on ACCEPT.

178397141-5ccc94a9-49ba-4ab7-ad67-b6d43a393659

- My Account Save confirmation dialog does not react to ENTER

178523236-db8bdc29-26f5-4178-881f-f14bfbeff36e

- Push Publish (The dialog does not gain focus immediately, you need to click on anywhere on the screen first, or use the tab button to navigate all the screen until the focus lands on the dialog)
- Content-Type (Behaves just fine). This is a perfect example of what most dialogs should behave like. Gain the focus and set it on the first active component or action,

In general, the dialog seems to execute a default action associated with the Key ENTER
But in general terms, the dialogs should gain focus by default and in most cases, they don't.

### PR DESCRIPTION
**Pull request by alfredo-dotcms**
Improvements on:
- Create Templates (Next Button does not react to ENTER)
- Apps / Import (The main dialog behaves correctly but once a file has been chosen and an invalid password is submitted) You'll see an invalid request dialog
- My Account Save confirmation dialog does not react to ENTER
- Push Publish (The dialog does not gain focus immediately, you need to click on anywhere on the screen first, or use the tab button to navigate all the screen until the focus lands on the dialog)

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
Original             |  Updated
:-------------------------:|:-------------------------:
** original screenshot **  |  ** updated screenshot **

### CODE REVIEW

**Review by jdotcms — APPROVED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-list/components/dot-template-selector/dot-template-selector.component.html [hunk 1]
```diff
@@ -1,7 +1,8 @@
-<div class="wrapper">
+<div class="wrapper" (keydown.enter)="onClick()">
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-templates/dot-template-list/components/dot-template-selector/dot-template-selector.component.html:1]: You should not be listening to the enter key but instead focus the button, that's for sematic html and specially a11y.

More here:

https://user-images.githubusercontent.com/751424/185980076-997fe265-c7d1-4de2-aa95-d69dd60ef388.mp4

Also, we should think a way to do this globally, somehow.

File: core-web/apps/dotcms-ui/src/app/view/components/_common/dot-alert-confirm/dot-alert-confirm.ts [hunk 2]
```diff
@@ -20,7 +22,8 @@ export class DotAlertConfirmComponent implements OnInit, OnDestroy {
             .pipe(takeUntil(this.destroy$))
             .subscribe(() => {
                 setTimeout(() => {
-                    this.confirmBtn.nativeElement.focus();
+                    const btn = this.confirmBtn || this.acceptBtn;
+                    btn.nativeElement.focus();
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/view/components/_common/dot-alert-confirm/dot-alert-confirm.ts:26]: More likely: https://angular.io/api/core/AfterViewInit

File: core-web/apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.html [hunk 3]
```diff
@@ -22,6 +22,7 @@
     <div class="field">
         <label class="form__label">{{ 'contenttypes.content.push_publish.filters' | dm }}: </label>
         <p-dropdown
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.html:24]: Did you try this? https://primefaces.org/primeng/autofocus or maybe the autofocus property of this component?

File: core-web/apps/dotcms-ui/src/app/view/components/_common/dot-alert-confirm/dot-alert-confirm.ts (hunk 2 above)
Comment (fmontes) [reply core-web/apps/dotcms-ui/src/app/view/components/_common/dot-alert-confirm/dot-alert-confirm.ts:26]: Or maybe https://primefaces.org/primeng/autofocus?

**Review by fmontes — CHANGES_REQUESTED** [review]
My first PR from vscode :D

File: core-web/apps/dotcms-ui/src/app/view/components/_common/dot-alert-confirm/dot-alert-confirm.ts (hunk 2 above)
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/view/components/_common/dot-alert-confirm/dot-alert-confirm.ts:26]: this alert dialog is loaded in the app in general, and the show/hide depends on a boolean Subject that enables it or not, so afterviewInit can not be used, neither the `autofocus`

File: core-web/apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.html (hunk 3 above)
Comment (alfredo-dotcms) [reply core-web/apps/dotcms-ui/src/app/view/components/_common/forms/dot-push-publish-form/dot-push-publish-form.component.html:24]: nice find!

**Review by hmoreras — APPROVED** [review]

**Review by fmontes — APPROVED** [review]

**Pull request merged**